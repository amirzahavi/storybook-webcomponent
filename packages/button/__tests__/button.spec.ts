
import {html} from 'lit-element';
import Button, { TAG_NAME } from "../lib/button";

describe(TAG_NAME, () => {
  it('should have click event', async () => {
    //Arrange
    const clickSpy = jest.fn();
    const btn = new Button();
    btn.addEventListener('click', clickSpy);
    //Act
    btn.click();
    //Assert
    expect(clickSpy).toHaveBeenCalledTimes(1);
  });

  it('should reflect slot', async () => {
    const rendered = html`<trax-button><span>test</span></trax-button>`;
    expect(rendered.getHTML()).toMatchSnapshot();
  });
});
