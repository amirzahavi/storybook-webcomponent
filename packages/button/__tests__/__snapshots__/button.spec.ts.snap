// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trax-button should reflect slot 1`] = `"<trax-button><span>test</span></trax-button>"`;
